ethereum/WebaverseERC721.sol

```diff
@@ -79,65 +82,118 @@ contract WebaverseERC721 is ERC721 {
+        tokenIdToBalance[tokenId] = SafeMath.add(
+            tokenIdToBalance[tokenId],
+            amount
+        );
 
-        require(erc20Contract.transferFrom(from, address(this), amount), "transfer failed");
+        require(
+            erc20Contract.transferFrom(from, address(this), amount),
+            "transfer failed"
+        );
     }
-    function unpack(address to, uint256 tokenId, uint256 amount) public {
+
+    /**
+     * @dev Unpack fungible tokens from this non-fungible token
+     * @param to Address of who is packing the token
+     * @param tokenId ID of the token
+     * @param amount Amount to unpack
+     */
+    function unpack(
+        address to,
+        uint256 tokenId,
+        uint256 amount
+    ) public {
         require(ownerOf(tokenId) == msg.sender, "not your token");
         require(tokenIdToBalance[tokenId] >= amount, "insufficient balance");
 
-        tokenIdToBalance[tokenId] = SafeMath.sub(tokenIdToBalance[tokenId], amount);
+        tokenIdToBalance[tokenId] = SafeMath.sub(
+            tokenIdToBalance[tokenId],
+            amount
+        );
 
         require(erc20Contract.transfer(to, amount), "transfer failed");
     }
 
-    function uint2str(uint _i) internal pure returns (string memory _uintAsString) {
-        if (_i == 0) {
-            return "0";
-        }
-        uint j = _i;
-        uint len;
-        while (j != 0) {
-            len++;
-            j /= 10;
-        }
-        bytes memory bstr = new bytes(len);
-        uint k = len - 1;
-        while (_i != 0) {
-            bstr[k--] = byte(uint8(48 + _i % 10));
-            _i /= 10;
-        }
-        return string(bstr);
-    }
-    
-    // 0x08E242bB06D85073e69222aF8273af419d19E4f6, 0x1, "lol", 1
-    function mint(address to, string memory hash, string memory name, string memory ext, string memory description, uint256 count) public {
-        require(!isSingleIssue, "wrong mint method called");
-        require(isPublicallyMintable || isAllowedMinter(msg.sender), "not allowed to mint");
-        require(bytes(hash).length > 0, "hash cannot be empty");
-        require(count > 0, "count must be greater than zero");
-        require(hashToTotalSupply[hash] == 0, "hash already exists");
+    /**
+     * @dev Mint one or more non-fungible tokens with this contract
+     * @param to Address of who is receiving the token on mint
+     * Example: 0x08E242bB06D85073e69222aF8273af419d19E4f6
+     * @param hash Hash of the file to mint
+     * Example: 0x1
+     * @param name Name of the token
+     * @param ext File extension of the token
+     * Example: "png"
+     * @param description Description of the token (set by user)
+     * @param count Number of tokens to mint (ie: 1)
+     */
+    function mint(
+        address to,
+        string memory hash,
+        string memory name,
+        string memory ext,
+        string memory description,
+        uint256 count
```

Comment:
This is where we should document that we only support a certain number of tokens being created at a time. The reason is that this loop over count eats a lot of gas.

This a non-obvious thing that deserves commenting.